File: src/main/java/compiler/parser/primitives/FloatParser.java
package compiler.parser.primitives;

import compiler.block.Block;
import compiler.block.primitives.FloatBlock;
import compiler.parser.Parser;
import compiler.tokenizer.Tokenizer;

public class FloatParser extends Parser<FloatBlock> {
    @Override
    public boolean shouldParse(String line) {
        return line.matches("float [a-zA-Z][a-zA-Z0-9]* [=] [0-9]+[.][0-9]*f");
    }

    @Override
    public FloatBlock parse(Block superBlock, Tokenizer tokenizer) {

        tokenizer.nextToken();  // skip float

        String name = tokenizer.nextToken().getToken();

        tokenizer.nextToken();

        String value = tokenizer.nextToken().getToken();
        tokenizer.nextToken();
        value += "." + tokenizer.nextToken().getToken();

        return new FloatBlock(superBlock,name,value);
    }
}

File: src/main/scala/compiler/parser/primitives/FloatParser.scala
package compiler.parser.primitives

import compiler.block.Block
import compiler.block.primitives.FloatBlock
import compiler.parser.Parser
import compiler.tokenizer.Tokenizer

class FloatParser extends Parser[FloatBlock] {
  def shouldParse(line: String): Boolean = line.matches("(val|var)[ ]+[a-zA-Z][a-zA-Z0-9]*[ ]*:[ ]*float[ ]*[=][ ]*[0-9]+[.][0-9]*f[ ]*")

  def parse(superBlock: Block, tokenizer: Tokenizer): FloatBlock = {
    val declaration: Boolean = tokenizer.nextToken.token == "val" // "val" or "var"
    val name: String = tokenizer.nextToken.token
    tokenizer.nextToken // skip ":"
    tokenizer.nextToken // skip "float"
    tokenizer.nextToken // skip "="
    var value: String = tokenizer.nextToken.token
    tokenizer.nextToken
    value += "." + tokenizer.nextToken.token
    new FloatBlock(superBlock, declaration,name, value)
  }
}
